mwdb/model/file.py

```diff
@@ -74,17 +74,86 @@ def get_or_create(cls, file, parent=None, metakeys=None, share_with=None):
 
         if is_new:
             file.stream.seek(0, os.SEEK_SET)
-            file.save(file_obj.get_path())
+            if app_config.mwdb.storage_provider == StorageProviderType.BLOB:
+                get_minio_client(
+                    app_config.mwdb.blob_storage_endpoint,
+                    app_config.mwdb.blob_storage_access_key,
+                    app_config.mwdb.blob_storage_secret_key,
+                    app_config.mwdb.blob_storage_region_name,
+                    app_config.mwdb.blob_storage_secure,
+                ).put_object(app_config.mwdb.blob_storage_bucket_name, file_obj._calculate_path(), file, file_size)
+            else:
+                file.save(file_obj._calculate_path())
 
         return file_obj, is_new
 
-    def get_path(self):
-        upload_root = app_config.mwdb.uploads_folder
+    def _calculate_path(self):
+        # upload_path must not be None
+        upload_path = app_config.mwdb.uploads_folder or ""
         sample_sha256 = self.sha256.lower()
-        # example: uploads/9/f/8/6/9f86d081884c7d659a2feaa0c55ad015a3bf4f1b2b0b822cd15d6c15b0f00a08
-        subdir_path = os.path.abspath(os.path.join(upload_root, *list(sample_sha256)[0:4]))
-        os.makedirs(subdir_path, mode=0o755, exist_ok=True)
-        return os.path.join(subdir_path, sample_sha256)
+        
+        if app_config.mwdb.hash_pathing:
+            # example: uploads/9/f/8/6/9f86d081884c7d659a2feaa0c55ad015a3bf4f1b2b0b822cd15d6c15b0f00a08
+            upload_path = os.path.join(upload_path, *list(sample_sha256)[0:4])
+    
+        if app_config.mwdb.storage_provider == StorageProviderType.DISK:
+            upload_path = os.path.abspath(upload_path)
+            os.makedirs(upload_path, mode=0o755, exist_ok=True)
+        return os.path.join(upload_path, sample_sha256)
+
+    # File API
+    _file_handle = None
```

Comment:
Move to the top of the file please, but is it actually necessary to have this as a class field? It looks like it can be a local variable in `iterate` method. Or am I wrong?